File: INF-E/src/main/webapp/scripts/controllers/supplierController.js
koekoek.controller('supplierController', ['$scope', '$http', '$routeParams', 'modalService', 'messageService', 'customerSupplierService',
                                           function($scope, $http, $routeParams, modalService, messageService, customerSupplierService) {
	$scope.suppliers = [];
	
	$scope.getSupplierList = function(){
		customerSupplierService.getSupplierList().then(function(result){
			$scope.suppliers = result.data;
	    });
	};

	$scope.getSupplierList();
	
	$scope.openModal = function(supplier){
		$scope.supplier = supplier;
		modalService.openModal("lg", "supplier", $scope);
	};
	
	$scope.dismissModal = function(){
		modalService.dismissModal();
		$scope.supplier = null;
	};
	
	$scope.deleteSupplier = function(supplierId){
		var promise = customerSupplierService.deleteSupplier(supplierId);
		promise.then(function(res){
			if(res.data.success == true){
				messageService.addSuccess("Leverancier verwijderd");
				$scope.getSupplierList();
			}else{
				messageService.addError(res.data.error);
				$scope.getSupplierList();
			}
	    });
	};
	
	$scope.saveSupplier = function(supplierName){
		var promise = customerSupplierService.saveSupplier(supplierName);
		
		promise.then(function(res){
			if(res.data.success == true){
				messageService.addSuccess("Leverancier opgeslagen.");
				$scope.dismissModal();
				$scope.getSupplierList();
			}else{
				messageService.addError(res.data.error);
				$scope.getSupplierList();
			}
	    });
		
	};
}]);

File: INF-E/src/main/java/nl/koekoek/controller/SupplierController.java
package nl.koekoek.controller;

import java.util.List;

import nl.koekoek.model.Supplier;
import nl.koekoek.service.SupplierService;
import nl.koekoek.support.RestResult;

import org.jarbframework.constraint.violation.UniqueKeyViolationException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

/**
 * Controller class for supplier.
 * @author Kevin van Leeuwen
 * 
 */
@Controller
@RequestMapping(value = "/supplier")
public class SupplierController {

    private final SupplierService supplierService;

    /**
     * Constructor for SupplierController object.
     * @param supplierService SupplierService object to use in the controller
     */
    @Autowired
    public SupplierController(SupplierService supplierService) {
        this.supplierService = supplierService;
    }

    /**
     * Returns all suppliers.
     * @return List with all suppliers
     */
    @RequestMapping(value = "/supplierList", method = RequestMethod.GET)
    @ResponseBody
    public List<Supplier> getSupplierList() {
        return supplierService.getAllSuppliers();
    }

    /**
     * Creates a supplier.
     * @param supplier new supplier object
     * @return the Restresult for a supplier
     */
    @RequestMapping(method = RequestMethod.POST)
    @ResponseBody
    public RestResult<Supplier> create(@RequestBody Supplier supplier) {
        try {
            return RestResult.success(supplierService.saveSupplier(supplier));
        } catch (UniqueKeyViolationException e) {
            return RestResult.error("Leverancier " + supplier.getName() + " bestaat al.");
        } catch (DataIntegrityViolationException d) {
            return RestResult.error("Leverancier " + supplier.getName() + " bestaat al.");
        }
    }

    /**
     * Updates a supplier.
     * @param newSupplier The supplier to be updated
     * @param supplier Supplier pathvariable
     * @return the Restresult for a supplier
     */
    @RequestMapping(value = "/{supplier}", method = RequestMethod.POST)
    @ResponseBody
    public RestResult<Supplier> edit(@RequestBody Supplier newSupplier, @PathVariable("supplier") Supplier supplier) {
        return RestResult.success(supplierService.saveSupplier(newSupplier));
    }

    /**
     * Deletes a supplier.
     * @param supplierId id of the supplier to be deleted
     * @return restresult 
     */
    @RequestMapping(value = "/delete/{supplierId}", method = RequestMethod.DELETE)
    @ResponseBody
    public RestResult<Supplier> delete(@PathVariable long supplierId) {
        try {
            Supplier supplier = supplierService.findById(supplierId);
            supplierService.deleteSupplier(supplier);
            return RestResult.success(supplier);
        } catch (DataIntegrityViolationException d) {
            return RestResult.error("Leverancier nog gekoppeld aan een server.");
        }
    }
}
